test_pull_requests

```diff
@@ -1184,8 +1304,56 @@ popd
         # comment to reflect the new state
         if submit_test_job
           delete_comment_with_prefix(id, base_repo, ACTION_PREFIX, comments)
-          extended_tests = get_extended_tests(req, comments, branch, settings)
-          submit_tests(repo_to_pull_request, base_repo, branch, comment_id, extended_tests, settings)
+          # Check for pretest_settings_key, so we might skip a round
+          # of tests prior to merge
+          preceding_test_settings = Properties['settings'][settings['pretest_settings_key']]
+          test_build_url = ''
+          quick_merge_comments = []
+          can_skip_merge_tests = false
+          if !preceding_test_settings.nil?
+            p_test_prefix = preceding_test_settings['test_prefix']
+            coreq_tests_synced_and_passed = repo_to_pull_request.all? do |repo, pull_request|
+              preceding_test_comment = get_comment_with_prefix(pull_request['number'], repo, p_test_prefix, (repo == base_repo ? comments : nil))
+              if preceding_test_comment.nil?
+                false
+              else
+                p_comment, p_test_build_url, p_base_commit, p_extended_tests = extract_bot_comment_metadata(preceding_test_comment['body'], preceding_test_settings)
+                p_base_commit = base_commit_tag_to_sha(p_base_commit)
+                can_skip_one = (p_comment =~ /^SUCCESS$/ &&
+                                p_base_commit == pull_request['base']['sha'])
+                if can_skip_one
+                  # Store pre-merge comments so we can carry the build url and base commit id through to SUCCESS
+                  quick_merge_comments << [pull_request['number'], repo, compose_bot_comment(settings['test_prefix'], 'Waiting:', :build_url => p_test_build_url, :base_commit => p_base_commit)]
```

Comment:
Ah.  So this is to add the build url.  You should be able to skip updating the comment right?  All the data is still in the test job to later update the merge comment with the correct build url.